src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleGroupSynchronizer.scala

```diff
@@ -0,0 +1,136 @@
+package org.broadinstitute.dsde.workbench.sam.google
+
+import akka.http.scaladsl.model.StatusCodes
+import cats.effect.IO
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.google.GoogleDirectoryDAO
+import org.broadinstitute.dsde.workbench.model._
+import org.broadinstitute.dsde.workbench.sam.directory.DirectoryDAO
+import org.broadinstitute.dsde.workbench.sam.model._
+import org.broadinstitute.dsde.workbench.sam.openam.AccessPolicyDAO
+import org.broadinstitute.dsde.workbench.sam._
+import org.broadinstitute.dsde.workbench.util.FutureSupport
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+class GoogleGroupSynchronizer(directoryDAO: DirectoryDAO,
```

Comment:
the functionality in this class was pulled from GoogleExtensions. It is separate because all these things now run on the background context, etc. and I didn't want that confusion in GoogleExetensions. Instead Boot figures it all out.